FreeRTOS/FreeRTOS-Kernel
# uxAutoReload replaced with xAutoReload to improve MISRA compliance

Description
-----------
The uxAutoReload value was documented to accept the values pdTRUE or pdFALSE.  The type of pdTRUE and pdFALSE is BaseType_t.  To fix this mismatch, the parameters to functions acception uxAutoReload were modified to accept a BaseType_t and a new function xTimerGetAutoReload was created to return a BaseType_t.  The legacy uxTimerGetAutoReload was left in place.

Test Steps
-----------
Coverity was executed to verify this change did not create new issues.
GCC was set with full errors/warnings/pedantic and compiled clean.
Demo CORTEX_M3_QEMU_GCC was run with no errors.

Related Issue
-----------
#164 


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
